DailyXKCD.js

```diff
@@ -33,49 +66,152 @@ Module.register("DailyXKCD", {
+        xkcd.id = "xkcdcontent";
         xkcd.src = this.dailyComic;
-        if (this.config.invertColors) {
+        if(this.config.invertColors){
             xkcd.setAttribute("style", "-webkit-filter: invert(100%);")
         }
+        comicWrapper.appendChild(xkcd);
 
+        wrapper.appendChild(comicWrapper);
 
-        if (this.config.title) wrapper.appendChild(title);
-
-        wrapper.appendChild(xkcd);
-
-        if (this.config.altText) {
+        if (this.config.showAltText) {
             var alt = document.createElement("div");
-            alt.className = "bright medium light";
+            alt.className = this.config.altTextFont;
             alt.innerHTML = this.dailyComicAlt;
             wrapper.appendChild(alt);
         }
 
-
         return wrapper;
     },
 
+    /* suspend()
+     * This method is called when a module is hidden.
+     */
+    suspend: function() {
+        this.scrollProgress = 0;
+
+        // reset to beginning if module is suspended, so we start at the top
+        var scrollable = document.getElementById("xkcdcontent");
+        scrollable.style.top = "0px";
+
+        for (var i = 0; i < this.autoIntervals.length; i++)
+        {
+            var current = this.autoIntervals[i];
+
+            if (current.interval)
+            {
+                clearInterval(current.interval);
+
+                current.interval = null;
+            }
+        }
+    },
+
+    /* resume()
+     * This method is called when a module is shown.
+     */
+    resume: function() {
+        for (var i = 0; i < this.autoIntervals.length; i++)
+        {
+            var current = this.autoIntervals[i];
+
+            if (!current.interval)
+            {
+                current.callback();
+
+                current.interval = setInterval(current.callback, current.time);
+            }
+        }
+    },
+
+    /* scrollComic
+     * Scrolls the comic down if needed
+     */
+    scrollComic: function() {
+        var scrollable = document.getElementById("xkcdcontent");
+
+        var height = scrollable.naturalHeight;
+
+        var top = 0;
+        if (this.config.limitComicHeight > 0 && height > this.config.limitComicHeight)
+        {
+            var currentHeight = this.scrollProgress * -this.config.limitComicHeight * this.config.scrollRatio;
+            var maxHeight = this.config.limitComicHeight - height;
+            top = Math.max(currentHeight, maxHeight);
+        }
+        scrollable.style.top = top + "px";
+        scrollable.style.height = height + "px";
+        if (top == this.config.limitComicHeight - height)
+        {
+            this.scrollProgress = -1;
+        }
+
+        this.scrollProgress += 1;
+    },
+
+    /* checkUserPresence(notification, payload, sender)
+     * Use this method to conveniently suspend your module when no user is present.
+     */
+    checkUserPresence: function(notification, payload, sender) {
+        if (sender && notification === "USER_PRESENCE") {
```

Comment:
The MMM-PIR-Sensor module ([github](https://github.com/paviro/MMM-PIR-Sensor) / [forum](https://forum.magicmirror.builders/topic/52/pir-sensor-put-your-mirror-to-sleep-if-not-used)).